src/components/imageUploader/imageUploader.js

```diff
@@ -124,6 +125,40 @@ import template from './imageUploader.template.html';
         page.querySelector('.btnBrowse').addEventListener('click', () => {
             page.querySelector('#uploadImage').click();
         });
+
+        page.querySelector('.btnDownloadFromUrl').addEventListener('click', () => {
+            showDownloadFromUrlPrompt(page);
+        });
+    }
+
+    function showDownloadFromUrlPrompt(page) {
+        import('../../components/prompt/prompt').then(async ({default: prompt}) => {
+            const url = await prompt({
+                title: globalize.translate('HeaderDownloadFromUrl'),
+                label: globalize.translate('LabelImageUrl'),
+                confirmText: globalize.translate('ButtonDownload')
+            });
+
+            try {
+                const response = await fetch(url);
+
+                if (response.ok) {
+                    const fileBlob = await response.blob();
+
+                    const extension = mime[response.headers.get('Content-Type') || 'image/jpeg'].extensions[0];
+
+                    const urlFileBlob = new File([fileBlob], `image.${extension}`, {
```

Comment:
Any idea if the File API will be supported by webOS/Tizen? It looks like older versions of Chrome may not https://caniuse.com/fileapi

Reply:
Good point.

Should we allow editing on TV at all, though? Seems like it'd be nicer to simply handle navigation/playback on TV and leave all the admin stuff out.

I haven't really tried to use the edit dialog in TV mode, but given how it looks, I doubt it's really usable :yum:

Reply:
I agree that a lot of stuff should probably just be disabled in TV mode, but if it's not this has the potential to crash the client.